lib/file_transfer_agent/local_util.js

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2015-2021 Snowflake Computing Inc. All rights reserved.
+ */
+
+var fs = require('fs');
+var path = require('path');
+const expandTilde = require('expand-tilde');
+const resultStatus = require('./file_util').resultStatus;
+
+/**
+ * Creates a local utility object.
+ *
+ * @returns {Object}
+ * @constructor
+ */
+function local_util()
+{
+  this.createClient = function (stageInfo, useAccelerateEndpoint)
+  {
+    return null;
+  }
+
+  /**
+  * Write file to upload.
+  *
+  * @param {Object} meta
+  *
+  * @returns {null}
+  */
+  this.uploadOneFileWithRetry = async function (meta)
+  {
+    await new Promise(function (resolve)
+    {
+      // Create stream object for reader and writer
+      var reader = fs.createReadStream(meta['realSrcFilePath']);
+      var output = path.join(meta['stageInfo']['location'], meta['dstFileName']);
+
+      // expand '~' and '~user' expressions
+      if (process.platform !== "win32")
+      {
+        output = expandTilde(output);
+      }
+
+      var writer = fs.createWriteStream(output);
```

Comment:
The path might not exist at this moment.

Reply:
Updated to create the directory if it doesn't exist.